### PR TITLE
Fix data mismatch in ScheduledJob causing celery worker to fail

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #3169
# What's Changed

- Add data migration to rename `ScheduledJob.kwargs["request"]["user"]` to `ScheduledJob.kwargs["request"]["_user_pk"]`

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/migrations/0054_scheduledjob_kwargs_request_user_change.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from django.db import migrations
+
+
+def rename_scheduled_job_kwargs_request_user(apps, schema_editor):
```
Comment (glennmatthews) [inline nautobot/extras/migrations/0054_scheduledjob_kwargs_request_user_change.py:4]: Maybe add a docstring and a cross-reference to #3105?

**Review by glennmatthews — APPROVED** [review]

**Review by bryanculver — APPROVED** [review]
```
[2023-01-26 18:21:44,588: INFO/MainProcess] celery@38e4e3ab4036 ready.
18:35:00.752 DEBUG   nautobot.core.celery __init__.py     nautobot_kombu_json_loads_hook() :
  Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,752: DEBUG/MainProcess] Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,755: INFO/MainProcess] Task nautobot.extras.jobs.scheduled_job_handler[af37fe31-9108-46ea-ad68-d3167d63456b] received
18:35:00.763 DEBUG   nautobot.core.celery __init__.py     nautobot_kombu_json_loads_hook() :
  Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,763: DEBUG/ForkPoolWorker-6] Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
18:35:00.824 DEBUG   nautobot.core.celery __init__.py                            default() :
  Performing nautobot serialization on <nautobot.utilities.utils.NautobotFakeRequest object at 0xffff9031bb10> for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,824: DEBUG/ForkPoolWorker-6] Performing nautobot serialization on <nautobot.utilities.utils.NautobotFakeRequest object at 0xffff9031bb10> for type nautobot.utilities.utils.NautobotFakeRequest
18:35:00.829 DEBUG   nautobot.core.celery __init__.py     nautobot_kombu_json_loads_hook() :
  Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,829: DEBUG/MainProcess] Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,829: INFO/MainProcess] Task nautobot.extras.jobs.run_job[682da9bd-2f41-41ea-b8b9-2a8f48abba92] received
[2023-01-26 18:35:00,829: INFO/ForkPoolWorker-6] Task nautobot.extras.jobs.scheduled_job_handler[af37fe31-9108-46ea-ad68-d3167d63456b] succeeded in 0.05919754199567251s: None
18:35:00.830 DEBUG   nautobot.core.celery __init__.py     nautobot_kombu_json_loads_hook() :
  Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
[2023-01-26 18:35:00,830: DEBUG/ForkPoolWorker-6] Performing nautobot deserialization for type nautobot.utilities.utils.NautobotFakeRequest
18:35:00.834 INFO    nautobot.jobs.ExampleJob jobs.py                                run_job() :
  Running job (commit=True)
[2023-01-26 18:35:00,834: INFO/ForkPoolWorker-6] Running job (commit=True)
18:35:00.836 INFO    nautobot.jobs.ExampleJob jobs.py                               _run_job() :
  job completed successfully
[2023-01-26 18:35:00,836: INFO/ForkPoolWorker-6] job completed successfully
18:35:00.846 INFO    nautobot.jobs.ExampleJob jobs.py                               _run_job() :
  Job completed in 0 minutes, 0.03 seconds
[2023-01-26 18:35:00,846: INFO/ForkPoolWorker-6] Job completed in 0 minutes, 0.03 seconds
[2023-01-26 18:35:00,847: INFO/ForkPoolWorker-6] Task nautobot.extras.jobs.run_job[682da9bd-2f41-41ea-b8b9-2a8f48abba92] succeeded in 0.016641958005493507s: True
```
:shipit:

**Comment by bryanculver** [comment]
Do we need to instruct users to purge their jobs already in the queue?

File: nautobot/extras/migrations/0054_scheduledjob_kwargs_request_user_change.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ScheduledJob = apps.get_model("extras", "ScheduledJob")
+    for sj in ScheduledJob.objects.all():
+        if "request" in sj.kwargs and sj.kwargs["request"].get("user", None) is not None:
```
Comment (gsnider2195) [inline nautobot/extras/migrations/0054_scheduledjob_kwargs_request_user_change.py:7]: Would `ScheduledJob.kwargs["request"]["user"]` ever be set to None? Should we just be looking for `if "request" in sj.kwargs and "user" in sj.kwargs["request"]:`

Comment (bryanculver) [reply nautobot/extras/migrations/0054_scheduledjob_kwargs_request_user_change.py]: Changes in 511498e resolved this and was validated.

**Review by jathanism — APPROVED** [review]

**Pull request merged**